migrate/cockroach.go

```diff
@@ -0,0 +1,183 @@
+package migrate
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/mattes/migrate"
+	"github.com/mattes/migrate/database/cockroachdb"
+	_ "github.com/mattes/migrate/source/file"
+
+	_ "github.com/lib/pq"
+)
+
+type Cockroach struct {
+	db *sql.DB
+
+	MigrationsPath string
+}
+
+type CockroachOptions struct {
+	*ConnectionOptions
+
+	MigrationTable string
+	Secure         bool
+	SSL            *CockroachSSL
+}
+
+type CockroachSSL struct {
+	CertPath, KeyPath, Mode, RootCert string
+}
+
+const (
+	DefaultHost           = "localhost"
+	DefaultUser           = "root"
+	DefaultPort           = "26257"
+	DefaultDatabase       = "service"
+	DefaultMigrationTable = "schema_migrations"
+)
+
+// Connection string concatenates the CockroachOptions down to a string,
+// applying defaults to options that were not set ready to be used in a
+// connection to the database
+func (co *CockroachOptions) ConnectionString() (string, error) {
+	// Prevent panics and just return the exact default upon nil options
+	if co.ConnectionOptions == nil {
+		return fmt.Sprintf(
+			"postgresql://%s@%s:%s/%s?sslmode=disable&x-migrations-table=%s",
+			DefaultUser, DefaultHost, DefaultPort, DefaultDatabase, DefaultMigrationTable,
+		), nil
+	}
+
+	// Start a string builder for the connection string
+	var conn strings.Builder
```

Comment:
As we're not going to be doing any of this in a critical path (or even live production software), I think `fmt.Sprintf` would be much easier to read!

Reply:
It may seem that way, but the way each option is optional with a default means there would either be one massive switch statement for all scenarios - unless you can suggest a better way i've not seen 😄

Reply:
We are most likely not going to use many options. a simple format string provided by the user for the few options the need is probably enough. That said: it works as it stands, so unless @codingconcepts corrects me, I think there's no harm in using this implementation.